### ISSUE TITLE
Improvements regarding "whitepaper" feature.

### ISSUE THREAD

**Issue by ConEstArthur** [Akylas/com.akylas.documentscanner#57]
I just wanted to let you know you're doing an amazing work on this project. It's baffling to me seeing the quality of the app as it's very intuitive, easy to use and free ?? How ? :)

To improve it further Few enhancements can be done to page scanning and overall quality of the scans.
I've tried every single (popular ones) document scanning app on play store. I don't know how vflat does it but when I scan a document on it, after the post processing, it eliminates all the marks of text that are on the other side of the page. 
The output looks way too clean. So it's may be related to "whitepaper" feature ? It can be improved further upon. 
After using "enhance" feature with "whitepaper" the difference becomes more apparent. Even more so with handwritten notes.

**Vflat**
![20231230_141857099](https://github.com/Akylas/com.akylas.documentscanner/assets/76774182/a9a400f5-c6e4-4df7-95dd-66c22d225e1e)

**Oss**
![1703984179191](https://github.com/Akylas/com.akylas.documentscanner/assets/76774182/7fba7fa8-1309-4781-88c7-3d21ae24602b)

**Comment by farfromrefug** [comment]
@ConEstArthur thanks for the kind words about the app. 

yes all can be improved ! can you share your original image before white paper post processing ?

it will be hard to get the same result as vflat but we never know...

**Comment by ConEstArthur** [comment]
> @ConEstArthur thanks for the kind words about the app.
> 
> yes all can be improved ! can you share your original image before white paper post processing ?
> 
> it will be hard to get the same result as vflat but we never know...

Ofcourse. Here it is -
![1703999816591](https://github.com/Akylas/com.akylas.documentscanner/assets/76774182/c6f90e5b-c04b-48ca-8dcf-a0cd9bd6df18)

Though It isn't that much of a big deal, I thought I should just let you know since this app has potential to become the best option for documents and making ebooks. :)

**Comment by mario-g98** [comment]
Whitepaper really needs improvement :/
![1721922174712 (1)](https://github.com/user-attachments/assets/dd546a41-e06c-471f-a4c3-3013f74590bb)
![1721922174712](https://github.com/user-attachments/assets/f50b9da8-0b4e-4529-a9e4-44895b31a434)

**Comment by farfromrefug** [comment]
@mario-g98 indeed i am not really happy with that algorithm. But for now i could not find a better one. I keep on searching but cant find what could work best.

**Comment by ConEstArthur** [comment]
> Whitepaper really needs improvement :/ !
Microsoft Lens works really well for this sample. (Not oss ofcourse)
I think vflat and lens are the examples we can draw inspiration from.
It'd surely be a complex venture to figure out all the algorithms.

**Comment by farfromrefug** [comment]
@ConEstArthur yes all have fast and accurate algorithm. But it is almost impossible to reverse engineer...
Our best bet would be to find an open source app which already did something like this, and maybe improve upon it

**Comment by ConEstArthur** [comment]
> Our best bet would be to find an open source app which already did something like this, and maybe improve upon it

Do paperless ngx or docspell comprise the feature similar to "Whiteboard" ?

**Comment by akumar-xyz** [comment]
Hey I want to take a crack at this in the next month or so. 

However, I think I need some guidance please

Are we using OpenCV imageproc for the current processing? What file should I be looking at for reference?

**Comment by jknockaert** [comment]
Perhaps an auto-threshold method should be implemented (or improved) to produce B/W output? I've been using a workflow of first deskewing, next resampling to 300dpi, subsequently converting to B/W using OTSU's method, and finally compressing with Group 4 algorithm, using greyscale scans as an input and producing compact and very good quality bitonal pdfs as an output.
